src/runCommand.js

```diff
@@ -1,8 +1,34 @@
-import { invalidate } from './invalidate'
+import { invalidate } from "./invalidate";
+import { query } from "./query";
+import omit from 'lodash/omit';
 
 export function runCommand(command, flatParams) {
-  return command.run(flatParams).then(v => {
-    invalidate(command.invalidates || {}, flatParams);
-    return v;
-  });
+  const noDeps = Object.keys(command.dependencies).length === 0;
+  if (noDeps) {
+    return command.run(flatParams).then(v => {
+      invalidate(command.invalidates, flatParams);
+      return v;
+    });
+  } else {
+    const depsToRun = omit(command.dependencies, Object.keys(flatParams))
+    return new Promise((resolve, reject) =>
+      query(depsToRun, flatParams)
+        .filter(event => !event.loading)
+        .subscribe(event => {
+          const queryResults = Object.keys(event.data).reduce(
+            (acc, k) => ({
+              ...acc,
+              [k]: event.data[k].data
+            }),
+            {}
+          );
+          resolve(
+            command.run({ ...queryResults, ...flatParams }).then(v => {
+              invalidate(command.invalidates, flatParams);
+              return v;
+            })
+          );
+        }, reject)
+    );
```

Comment:
this is still missing a `subscription.unsubscribe()` before resolving

Reply:
I added a `first` after the filter, which should mean "unsubscribe after the first event where no more dependencies are loading".

Reply:
nice, much better than imperatively unsubscribing 👏